src/avk.cpp

```diff
@@ -4527,9 +4527,40 @@ namespace avk
 #endif
 	}
 
+	// Unfortunately C++20 does not have support to easily convert a range into a vector through chaining, so we have to
+	// either pass the ranges begin and end members to the vector constructor every time or implement a helper ourself
+	// until something similar is standardized (There is a to<Container> proposal for C++23).
+	// This helper implementation converts a range into a vector by piping the range into to_vector():
+	//
+	// auto nums = std::vector{1, 2, 3, 4};
+	// auto num_strings = nums | std::ranges::transform([](auto num) { return std::to_string(num); }) | to_vector();
+	// 
+	namespace to_vector_impl
+	{
+		// Helper type to find the operator| overload below
+		struct to_vector_helper {};
+
+		// Operator| overload to convert a range into a vector
+		template <std::ranges::range R>
+		auto operator|(R&& r, to_vector_helper)

```

Comment:
Did you come up with this by yourself? If not, a citation would be appropriate.

Reply:
Fair point. While I'm not sure where I have first heard of this little trick since I have already used it on a different project years ago with `operator<<`, I probably would not have immediately thought about it without beeing reminded in a _stackoverflow_ thread while researching `ranges` alternatives to `cpplinq`